samples/pipeline/masstransit-messages/Core_8/NServiceBusSubscriber/MassTransitIngestBehavior.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NServiceBus.Pipeline;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NServiceBusSubscriber
+{
+    using System.Globalization;
+
+    #region Behavior
+    public class MassTransitIngestBehavior : Behavior<IIncomingPhysicalMessageContext>
+    {
+        public override Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
+        {
+            if (!context.MessageHeaders.TryGetValue("Content-Type", out var contentType) || contentType != "application/vnd.masstransit+json")
+            {
+                return next();
+            }
+
+            var envelope = DeserializeMassTransitPayload(context);
+
+            AddHeaderIfExists(context, envelope, "messageId", NServiceBus.Headers.MessageId);
+            AddHeaderIfExists(context, envelope, "conversationId", NServiceBus.Headers.ConversationId);
+            AddHeaderIfExists(context, envelope, "correlationId", NServiceBus.Headers.CorrelationId);
+            AddHeaderIfExists(context, envelope, "sourceAddress", "MassTransit.SourceAddress");
+            AddHeaderIfExists(context, envelope, "destinationAddress", "MassTransit.DestinationAddress");
+            AddHeaderIfExists(context, envelope, "sentTime", NServiceBus.Headers.TimeSent, sentTime =>
+            {
+                var parsed = DateTime.Parse(sentTime, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AllowWhiteSpaces);
```

Comment:
this had to be adjusted since it threw an exception when parsing without further arguments.